### DISCUSSION TITLE
Slow documentation build

### DISCUSSION THREAD

**Discussion by nijel** [pradyunsg/furo#845] [Potential Issue]
I've recently switched [translate-toolkit documentation](https://docs.translatehouse.org/projects/translate-toolkit/en/latest/) to furo (in https://github.com/translate/translate/pull/5338).

What I didn't notice when doing the change is that the documentation build time went from ~1 minute to ~5 minutes just because of changing the theme. Is something like this expected? What would be the best approach to find what causes the slowdown?

**Comment by nineteendo** [comment]
The sidebar maybe? Furo is intended for smaller documentation sets.

**Reply by nijel** [reply]
I've tried removing some documents from toctree and that doesn't seem to have any visible impact. Completely disabling toctree rendering makes the difference, though (I've modified `_compute_navigation_tree` to not call `toctree` at all).

PS: I'm using Furo for [Weblate documentation](https://docs.weblate.org/en/latest/) for years and it builds in ~30 seconds while it is not really small (contains 152 rst documents, translate-toolkit has 259).

**Reply by nijel** [reply]
I revisited this today, and it turns out to be caused by the API reference (for example https://docs.translatehouse.org/projects/translate-toolkit/en/latest/api/convert.html) where the page content is massive. I thought that only the size of the actual TOC would affect the performance, not the size of the individual documents…
